assets/app/features/upgrade/controllers/upgrade7-openstack-services.controller.js

```diff
@@ -48,19 +51,16 @@
                     
                 ).finally(function() {
                     // Either on sucess or failure, the openStackServices has been completed.
-                    vm.openStackServices.completed = true;
-
-                    // Update openStackServices validity
-                    
-
+                    vm.openStackServices.completed = true;              
+                    vm.openStackServices.running = false;
```

Comment:
Adding more documentation won't hurt but this problem is not strictly spinner related. It's about understanding when the action is "running". When there are multiple asynchronous actions chained together then you need to be careful when the actual "run" is finished as there are multiple branches with possible errors and successes.